release-management/create-changelog/readme.md

```diff
@@ -5,27 +5,57 @@ This script generated a changelog based on merged pull requests. To generate a
 changelog for a given version, run the script with the git branch name, start
 date and optionally end date as arguments.
 
-
-Example for 4.x
+Example for 14.x
 ---------------
 
 ```
-% python changelog.py r/4.x 2018-03-29                                                                                                                       (git)-[master] [148] 
-- [MH-12923 serviceregistry initializes a db connection twice
-  ](https://github.com/opencast/opencast/pull/267)
-...
+- [[#4945](https://github.com/opencast/opencast/pull/4945)] -
+  Drop orphan statistics database index
 ```
 
-For version 4.0 you would additionally add the changes listed by
-
-    % python3 changelog.py develop 2017-09-22 2018-04-03
+Changelog for N.x version
+-------------------------
 
+    python changelog.py r/N.x <N.x-branch-cut-date>
 
-Changelog for x.0 version
+Changelog for N.0 version
 -------------------------
 
 Since these versions are developed on both `develop` and their specific release
 branched, two requests need to be made and merged:
 
-    % python changelog.py develop <begin-of-development> <x.0-branch-cut-date>
-    % python changelog.py r/5.x <begin-of-development>
+    python changelog.py develop <begin-of-development> <N.x-branch-cut-date>
+    python changelog.py r/N.x <N.x-branch-cut-date>
+
+Note that the Github API may generate duplicate entries between the two lists depending on dates and timezones.
+
+Dates
+-----
+Computing the dates can be annoying.  You need to find the earliest commit belonging to various combinations of branches.
+
+    git log --pretty=%as $(diff -u <(git rev-list --first-parent r/N.x) <(git rev-list --first-parent develop) | sed -ne 's/^ //p' | head -1)
```

Comment:
I think this should be this? Also the same in the example blow.
```suggestion
    git log --pretty=%as $(diff -u <(git rev-list --first-parent r/N.x) <(git rev-list --first-parent develop) | sed -ne 's/^ //p') | head -1
```

Reply:
The the initial `diff` spits out a *list* of commits, which is why the `head` needs to be inside the `$()`.  Otherwise you end up with a list of dates, rather than just the one you want - the first one

Reply:
ok makes sense, but I will get a list of dates regardless so I think you need a second `head` at the end. The example below boils down to `git log --pretty=%as 399d712955b506624d31abb19863339a558e9bc1`, which will return a list of dates on my system.